parsers/cxx/cxx_parser_tokenizer.c

```diff
@@ -0,0 +1,623 @@
+		t->bFollowedBySpace = FALSE;
+		return FALSE;
+	}
+
+	unsigned int uInfo = (g_cxx.iChar < 0x80) ? g_aCharTable[g_cxx.iChar].uType : 0;
+	
+	//printf("Char %c %02x info %u\n",g_cxx.iChar,g_cxx.iChar,uInfo);
+
+	if(uInfo & CXXCharTypeStartOfIdentifier)
+	{
+		// word
+		t->eType = CXXTokenTypeIdentifier;
+		t->bFollowedBySpace = FALSE;
+
+		vStringPut(t->pszWord,g_cxx.iChar);
+
+		// special case for tile, which may actually be an operator
+		if(g_cxx.iChar == '~')
+		{
+			// may be followed by space!
+			g_cxx.iChar = cppGetc();
+			if(isspace(g_cxx.iChar))
+			{
+				t->bFollowedBySpace = TRUE;
+				g_cxx.iChar = cppGetc();
+				while(isspace(g_cxx.iChar))
+					g_cxx.iChar = cppGetc();
+			}
+
+			// non space
+			uInfo = (g_cxx.iChar < 0x80) ? g_aCharTable[g_cxx.iChar].uType : 0;
+			if(!(uInfo & CXXCharTypeStartOfIdentifier))
+			{
+				// this is not an identifier after all
+				t->eType = CXXTokenTypeOperator;
+				if((!t->bFollowedBySpace) && g_cxx.iChar == '=')
+				{
+					// make ~= single token so it's not handled as a separate assignment
+					vStringPut(t->pszWord,g_cxx.iChar);
+					g_cxx.iChar = cppGetc();
+					t->bFollowedBySpace = isspace(g_cxx.iChar);
+				}
+				return TRUE;
+			}
+		} else {
+			g_cxx.iChar = cppGetc();
+		}
+
+		for(;;)
+		{
+			uInfo = (g_cxx.iChar < 0x80) ? g_aCharTable[g_cxx.iChar].uType : 0;
+			if(!(uInfo & CXXCharTypePartOfIdentifier))
+				break;
+			vStringPut(t->pszWord,g_cxx.iChar);
+			g_cxx.iChar = cppGetc();
+		}
+
+		int iCXXKeyword = lookupKeyword(t->pszWord->buffer,g_cxx.eLanguage); 
+		if(iCXXKeyword >= 0)
+		{
+			if((iCXXKeyword == CXXKeywordFINAL) && (!g_cxx.bParsingClassStructOrUnionDeclaration))
+			{
+				t->eType = CXXTokenTypeIdentifier;
+			} else {
+				t->eType = CXXTokenTypeKeyword;
+				t->eKeyword = (enum CXXKeyword)iCXXKeyword;
+			}
+		} else {
+			boolean bIgnoreParens = FALSE;
+			const char * szReplacement = NULL;
+			if(isIgnoreToken(vStringValue(t->pszWord),&bIgnoreParens,&szReplacement))
+			{
+				CXX_DEBUG_PRINT("Ignore token %s",vStringValue(t->pszWord));
+				// FIXME: Handle ignore parens!
+				if(szReplacement && *szReplacement)
+				{
+					vStringClear(t->pszWord);
+					vStringCatS(t->pszWord,szReplacement);
+				} else {
+					// skip
+					cxxTokenChainDestroyLast(g_cxx.pTokenChain);
+					return cxxParserParseNextToken();
+				}
+			}
+		}
+
+		t->bFollowedBySpace = t->bFollowedBySpace | isspace(g_cxx.iChar);
+
+		return TRUE;
+	}
+	
+	if(g_cxx.iChar == '-')
+	{
+		// special case for pointer 
+		vStringPut(t->pszWord,g_cxx.iChar);
+		g_cxx.iChar = cppGetc();
+		if(g_cxx.iChar == '>')
+		{
+			t->eType = CXXTokenTypePointerOperator;
+			vStringPut(t->pszWord,g_cxx.iChar);
+			g_cxx.iChar = cppGetc();
+		} else {
+			t->eType = CXXTokenTypeOperator;
+			if(g_cxx.iChar == '-')
+			{
+				vStringPut(t->pszWord,g_cxx.iChar);
+				g_cxx.iChar = cppGetc();
+			}
+		}
+		t->bFollowedBySpace = isspace(g_cxx.iChar);
+		return TRUE;
+	}
+	
+	if(g_cxx.iChar == '"')
```

Comment:
as you use `cppGetc()`, isn't this dead code?  it should be returning `STRING_SYMBOL`, not bare quotes